### PR TITLE
HHH-16743 StackOverflowError when loading a ManyToOne whith @Proxy(lazy=false)

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-16743

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializerBuilder.java [hunk 1]
```diff
@@ -134,6 +134,11 @@ else if ( creationState.isDynamicInstantiation() ) {
 			if ( cacheAccess != null ) {
 				// Do batch initialization instead of batch loading if the parent entity is cacheable
 				// to avoid putting entity state into the cache at a point when the association is not yet set
+				if ( entityPersister.getRepresentationStrategy()
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializerBuilder.java:137]: Should this rather be part of the first if condition in this method?

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializerBuilder.java:137]: not really, it fine if it is a BATCH_LOAD but probably we have to do this check also at line 108 and 123 where we may return BATCH_INITIALIZE.

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializerBuilder.java]: @beikov what do you think of the new change?

Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializerBuilder.java]: LGTM

**Pull request merged**